modules/swagger-codegen/src/main/resources/php/api.mustache

```diff
@@ -122,22 +124,38 @@ class {{classname}} {
         // for HTTP post (form)
         $httpBody = $formParams;
       }
-
-      // authentication setting, if any
-      $authSettings = array({{#authMethods}}'{{name}}'{{#hasMore}}, {{/hasMore}}{{/authMethods}});
-
+      {{#authMethods}}{{#isApiKey}}
```

Comment:
Thanks for sharing the use case with me (first time I heard of such use case). I think you've a valid reason to keep the authentication logic out of the ApiClient so please keep your approach as the change should be transparent to the developers using the library.

For other clients, we'll consider doing the same.